src/libsyntax/parse/token.rs

```diff
@@ -629,6 +539,105 @@ impl Token {
             _ => None,
         }
     }
+
+    /// Returns `true` if the token can appear at the start of an expression.
+    crate fn can_begin_expr(&self) -> bool {
```

Comment:
From a diff perspective I would have preferred to just move around `impl Token` and `impl TokenKind` blocks.

Reply:
Do you mean
```rust
impl Token {
    ...
}

impl TokenKind {
    ...
}
```
?
I've just tried it and the total diff doesn't look much better (the code still looks like it's moved around rather than edited, just in the opposite direction).

Or do you mean multiple `impl Token`/`impl TokenKind` blocks alternating between each other?

Reply:
> Or do you mean multiple impl Token/impl TokenKind blocks alternating between each other?

that